zerver/lib/actions.py

```diff
@@ -2335,7 +2335,13 @@ def check_message(sender: UserProfile, client: Client, addressee: Addressee,
             stream = addressee.stream()
         assert stream is not None
 
-        recipient = stream.recipient
+        # We create a recipient object that matches what is
+        # already in the database.
+        recipient = Recipient(
+            id=stream.recipient_id,
+            type_id=stream.id,
+            type=Recipient.STREAM,
+        )
```

Comment:
Hmm, I don't see a specific way that this would value as we don't try to `.save()` the `message.recipient` field, but this seems like we're violating some sort of abstraction barrier.  Could we just do a `.select_related("recipient")` in the functions that get a Stream object for these code paths?  

There are several code paths involved but I think all of them end up calling `get_stream_by_id`; maybe we should just make that `.select_related("recipient")`?  It'd be basically free to do so and would mean we can expect to access recipient objects freely in other code paths as well.